src/data/validate_simulator.py

```
@@ -0,0 +1,517 @@
+    # 1. 행동 감쇠 검증
+    decay_stats = compute_pre_churn_decay(events, customers)
+    decay_ok = 0.4 <= decay_stats["ratio"] <= 0.7
+    print(f"\n[1] 이탈 전 30일 vs 직전 30일 비율: {decay_stats['ratio']:.3f}")
+    print(f"    (D-30~D-1 평균 이벤트: {decay_stats['mean_events_d30_d1']:.2f}  "
+          f"D-60~D-31: {decay_stats['mean_events_d60_d31']:.2f})")
+    print(f"    검증 대상 고객: {decay_stats['n_customers']:,}명")
+    print(f"    목표: 0.4~0.7  결과: {'PASS' if decay_ok else 'FAIL'}")
+
+    # 2. Top 5 신호
+    top5 = extract_top5_signals(events, customers)
+    print(f"\n[2] 이탈 직전 Top 5 행동 신호:")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)